### PR TITLE
feat: bypass cache check with forceFetch param

### PR DESCRIPTION
**Pull request by Fyko**
**Please describe the changes this PR makes and why it should be merged:**
This PR adds the parameter `forceFetch` on various functions that would check a manager's cache before falling back to requesting the API. If true, the cache check will be bypassed and hit the API. If undefined, the parameter will default to `false`.

Effected functions include:
- Managers
  - `ChannelManager#fetch`
  - `GuildMemberManager#fetch` (added to the options object)
  - `MessageManager#fetch`
  - `RoleManager#fetch`
  - `UserManager#fetch`
- Structures
  - `Channel#fetch`
  - `DMChannel#fetch` (relies on `User#createDM`)
  - `GuildMember#fetch`
  - `Message#fetch`
  - ~~`MessageReaction#fetch`~~ may not be necessary here
  - `User#fetchFlags`
  - `User#fetch`
  - `User#createDM` (underlying support for `DMChannel#fetch`)

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**

- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by BannerBomb** [comment]
I just want to ask but wouldn't this be redundant to do?

I am just trying to figure out if there's a valid reason to skip the cache. Because for what I know the reason discord.js checks the cache first is to prevent the client having to send another request to the API. Plus you could get the same effect by manually deleting the object from the cache then fetching which also won't give any benefits for doing so.

**Comment by advaith1** [comment]
> I am just trying to figure out if there's a valid reason to skip the cache.

the cache can be outdated

File: src/structures/User.js [hunk 1]
```diff
@@ -222,11 +222,15 @@ class User extends Base {
 
   /**
    * Creates a DM channel between the client and the user.
+   * @param {boolean} [forceFetch=false] Whether to skip the cache check and request the API
    * @returns {Promise<DMChannel>}
    */
-  async createDM() {
-    const { dmChannel } = this;
-    if (dmChannel && !dmChannel.partial) return dmChannel;
+  async createDM(forceFetch = false) {
```
Comment (vladfrangu) [inline src/structures/User.js:228]: This might be better named as forceCreate?

**Comment by iCrawl** [comment]
Additionally this needs a rebase.

File: src/structures/User.js (hunk 1 above)
Comment (Fyko) [reply src/structures/User.js]: impl with https://github.com/discordjs/discord.js/pull/4592/commits/5daa176142c067712cb9aeaaa52733083a8adb56
cc: @kyranet

File: src/managers/MessageManager.js [hunk 2]
```diff
@@ -62,8 +63,8 @@ class MessageManager extends BaseManager {
    *   .then(messages => console.log(`${messages.filter(m => m.author.id === '84484653687267328').size} messages`))
    *   .catch(console.error);
    */
-  fetch(message, cache = true) {
-    return typeof message === 'string' ? this._fetchId(message, cache) : this._fetchMany(message, cache);
+  fetch(message, cache = true, force = true) {
```
Comment (kyranet) [inline src/managers/MessageManager.js:66]: This default (`true`) does not match the default in the JSDocs (line 48, documented as `false`).

**Pull request merged**